core/new-gui/src/app/dashboard/component/feature-container/user-project-section/user-project-section.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, OnInit } from "@angular/core";
+import { UserProjectService } from "../../../service/user-project/user-project.service";
+import { Project } from "../../../type/project";
+import { Router } from "@angular/router";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { NzMessageService } from "ng-zorro-antd/message";
+
+export const ROUTER_PROJECT_BASE_URL = "/dashboard/project";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-user-project-section",
+  templateUrl: "./user-project-section.component.html",
+  styleUrls: ["./user-project-section.component.scss"]
+})
+export class UserProjectSectionComponent implements OnInit {
+  public projectEntries: Project[] = [];
+  public projectEntriesIsEditingName: number[] = [];
+  public createButtonIsClicked: boolean = false;
+  public createProjectName: string = "";
+
+  constructor(
+    private userProjectService: UserProjectService,
+    private router: Router,
+    private message: NzMessageService
+  ) { 
+  }
+
+  ngOnInit(): void {
+    this.getProjectArray();
+  }
+
+  private getProjectArray() {
+    
+    this.userProjectService
+      .retrieveProjectList()
+      .pipe(untilDestroyed(this))
+      .subscribe(projectEntries => {
+        this.projectEntries = projectEntries;
+      });
+  }
+
+  /**
+   * navigate to individual project page
+   */
+  public jumpToProject({ pid }: Project): void {
+    this.router.navigate([`${ROUTER_PROJECT_BASE_URL}/${pid}`]).then(null);
+  }
+
+  public removeEditStatus(pid : number): void {
+    this.projectEntriesIsEditingName = this.projectEntriesIsEditingName.filter(index => index != pid);
+  }
+
+  public saveProjectName(project: Project, newName: string, index: number): void {
+    // nothing happens if name is the same
+    if (project.name === newName) {
+      this.removeEditStatus(project.pid);
+    } else if (this.projectEntries.filter(p => project.pid !== p.pid && p.name === newName).length > 0) {
+      // checks the projects belonging to user to see if this will create duplicate name
+
+      // show error message and don"t call backend
+      this.message.create("error", `Project named: ${newName} already exists`);
```

Comment:
I see we have some fronted validation on `project-name`. please make this validation a utility function and reuse the logic on `create project`, `update project name`, etc. We still need the backend validation logic even though we have the front-end one.

Reply:
done